src/ops/view.rs

```diff
@@ -12,6 +15,150 @@ use cargo::{
 
 use super::style::{ERROR, HEADER, LITERAL, NOP, NOTE, WARN};
 
+#[allow(unused)]
+#[derive(PartialEq, Eq)]
+enum Hyperlink<'a> {
+    Crate(InternedString, Option<&'a semver::Version>),
+    Documentation(InternedString, Option<&'a semver::Version>),
+    Keyword(&'a str),
+    // Intrinsic link, for `crates.io`, `docs.rs` and etc.
+    Intrinsic(&'a str),
+    // External link, for `github` and etc.
+    External(&'a str),
+}
+
+impl<'a> Hyperlink<'a> {
+    // Judge about whether current terminal supports hyper link or not.
+    // Caution: If running a test, this will always return `false`.
+    fn is_terminal_supports() -> bool {
+        static TERMINAL_SUPPORT_HYPER_LINK: OnceLock<bool> = OnceLock::new();
+
+        *TERMINAL_SUPPORT_HYPER_LINK.get_or_init(|| {
+            // Copied from https://github.com/rust-lang/cargo/blob/a8d72c675ee52dd57f0d8f2bae6655913c15b2fb/src/cargo/core/shell.rs#L568.
+            #[allow(clippy::disallowed_methods)]
+            // We are reading the state of the system, not config.
+            if std::env::var_os("TERM_PROGRAM").as_deref()
+                == Some(std::ffi::OsStr::new("iTerm.app"))
+            {
+                // Override `supports_hyperlinks` as we have an unknown incompatibility with iTerm2
+                return false;
+            }
+
+            supports_hyperlinks::supports_hyperlinks()
+        })
+    }
+
+    // Calculate the text which can be a hyperlink on terminal.
+    // Refer to: https://gist.github.com/egmontkob/eb114294efbcd5adb1944c9f3cb5feda
+    fn format_terminal_link<'b>(
+        link: &'b str,
+        text: Option<&'b str>,
+        id: Option<&'b str>,
+    ) -> String {
+        let text = text.unwrap_or(link);
+
+        if let Some(id) = id {
+            format!("\u{1b}]8;id={id};{link}\u{1b}\\{text}\u{1b}]8;;\u{1b}\\")
+        } else {
+            format!("\u{1b}]8;;{link}\u{1b}\\{text}\u{1b}]8;;\u{1b}\\")
+        }
+    }
+
+    // Calculate the related link.
+    fn as_link_lossy(&self) -> Cow<'a, str> {
+        match self {
+            Hyperlink::Crate(name, version) => Cow::Owned(if let Some(version) = version {
+                format!("https://crates.io/crates/{name}/{version}")
+            } else {
+                format!("https://crates.io/crates/{name}")
+            }),
+            Hyperlink::Documentation(name, version) => Cow::Owned(if let Some(version) = version {
+                format!("https://docs.rs/{name}/{version}")
+            } else {
+                format!("https://docs.rs/{name}")
+            }),
+            Hyperlink::Keyword(keyword) => {
+                Cow::Owned(format!("https://crates.io/keywords/{keyword}"))
+            }
+            Hyperlink::Intrinsic(link) => Cow::Borrowed(*link),
+            Hyperlink::External(link) => Cow::Borrowed(*link),
+        }
```

Comment:
Why is knowledge of each type of link centralized here?  And whats the difference between Intrinsic and External?

Reply:
Hmmm...

The `Intrinsic` means for the links which refer to the `crate.io` or `docs.rs` sites.

The `External` is the opposite against the `Intrinsic`.

These two are both be thought as a `newType` to identify the links.

Reply:
But why is it centralized?  I feel like this adds unnecessary extra layers.  Look up in Cargo how we are using `out_hyperlink` to see how we do links normally.

Reply:
Hi Ed~

I had removed these codes~